src/main/java/net/glowstone/entity/GlowAnimal.java

```diff
@@ -31,4 +38,36 @@ public GlowAnimal(Location location, EntityType type, double maxHealth) {
     protected int getAmbientDelay() {
         return 120;
     }
+
+    @Override
+    public boolean entityInteract(GlowPlayer player, InteractEntityMessage message) {
+        super.entityInteract(player, message);
+
+        if (message.getAction() == InteractEntityMessage.Action.INTERACT.ordinal()) {
+            ItemStack item = InventoryUtil
+                    .itemOrEmpty(player.getInventory().getItem(message.getHandSlot()));
+
+            if (player.getGameMode().equals(GameMode.SPECTATOR)
+                    || InventoryUtil.isEmpty(item)) {
+                return false;
+            }
+
+            if (GameMode.CREATIVE != player.getGameMode()
+                    && getBreedingFoods().contains(item.getType())) {
+                // TODO set love mode if possible and spawn particles
+                // TODO heal
+                player.getInventory().consumeItemInMainHand();
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Returns a set containing the breeding foods for the current animal.
+     * @return a set containing Material
+     */
+    public Set<Material> getBreedingFoods() {
+        return EnumSet.noneOf(Material.class);
```

Comment:
Perhaps cache that set in a static field?

Reply:
This default is almost never used, because most of all the subclasses override this method. That's why i didn't cache this one. Fixed now ;)